[PE-4174] Added Enterprise Global Navigation example to Sidenav component

janibolkvadze opened the pull request:
### Commit summary

The first implementation of Enterprise Global Navigation 3rd level tabs.

- [x] Implemented 3rd level navigation with animation
- [x] Added an example to /javascript/sidenav view
- [x] Fixed some JS linting errors

Thoughts about the approach, class namings, etc? @mattnickles @jllr @dani-cl-madrid 

https://ctl.atlassian.net/browse/PE-4174

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/538/ for more details.

janibolkvadze commented:
- [x] Updated Enterprise Global Navigation Sidenav design by overriding default properties
- [x] Fixed sliding border bug on inversed sidenavs
- [x] Improved closing behavior when subtabs are clicked
- [x] Improved activating of menu items when subtabs are visible

Still pending:
- [ ] Changing sliding border color from green to white when sliding from active menu item to non-active (and vice versa)

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/540/ for more details.

mattnickles commented on src/chi/javascript/components/sidenav.js:
```diff
@@ -192,10 +198,27 @@ class Sidenav extends Component {
+        this.close(menuItemLink);
+      }
 
       if (drawerMenuItem) {
```
Can the Drawer menu item list function as a toggle? If expanded, clicking again would collapse

mattnickles replied:
@janibolkvadze Collapsing effect is very slow. Can we eliminate that delay? We should also probably remove the `-active` class after collapsing.

mattnickles commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +404,114 @@ $sidenav-lg-width: 12.5rem;
+  &.-global-nav {
+    &.-inverse {
+      .m-sidenav__content {
+        border-right: 0;
```
Line 152 - Background color should be grey 100, not black

mattnickles commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +404,114 @@ $sidenav-lg-width: 12.5rem;
+        .m-drawer {
+          &.-active {
+            .m-drawer__header {
+              background-color: set-color(grey, 100);
```
Background color should be grey 100 with a 50% alpha

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/541/ for more details.

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/545/ for more details.

mattnickles commented:
@janibolkvadze If Menu Item 1 is active and Menu Item 2 is clicked, I see we're applying the class `unselected` to Menu Item 1. However, if no secondary link within Menu Item 2 was clicked and the **drawer is closed**, we should probably remove the `unselected` class from Menu Item 1 so it appears as the active item again. What do you think?

jllr commented:
@dani-cl-madrid can you take a look at this PR too? Thanks in advance

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -116,6 +116,16 @@ $sidenav-lg-width: 12.5rem;
+        overflow: hidden;
+        transition: height 0.075s ease-out;
+
+        &.visible {
```
@janibolkvadze we usually use - (dash) for this kind of classes (ie: -active). Please add it.

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+                  a {
+                    background-color: rgba(set-color(mint, 40), 0.08);
+
+                    &.unselected {
```
@janibolkvadze we usually use - (dash) for this kind of classes (ie: -active). Please add it.

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+              background-color: rgba(set-color(grey, 100), 0.5);
+              color: set-color(grey, 40);
+              display: flex;
+              font-size: 16px;
```
@janibolkvadze please remove this property and add it to .a-drawer__title and convert it to rem

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+
+              .a-drawer__title {
+                font-weight: 600;
+                left: 1.25rem;
```
@janibolkvadze this property is not necessary. Please remove it.

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+
+                li {
+                  border-bottom: 0;
+                  line-height: 20px;
```
@janibolkvadze please convert it to rem.

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+                        li {
+                          a {
+                            &.a-sidenav__drawer-item-tab {
+                              font-size: 13px;
```
@janibolkvadze please convert it to rem.

jllr commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+                            &.a-sidenav__drawer-item-tab {
+                              font-size: 13px;
+                              font-weight: 400;
+                              line-height: 20px;
```
@janibolkvadze please convert it to rem.

mattnickles commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+
+              .m-sidenav__drawer-list {
+                border-top: 0;
+                padding-top: 0.5rem;
```
Let's remove this `padding-top: 0.5rem;`

mattnickles commented on src/chi/components/sidebar/sidenav.scss:
```diff
@@ -376,6 +398,134 @@ $sidenav-lg-width: 12.5rem;
+                              padding-bottom: 0.375rem;
+                              padding-left: 3rem;
+                              padding-top: 0.375rem;
+                            }
```
In their design a hover state is defined on `m-sidenav__drawer-item-list` tab items. Let's add:
```
&:hover {
   background-color: set-color(grey, 70);
}
```

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/560/ for more details.

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/562/ for more details.

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/575/ for more details.

allyourbotsarebelongtous commented:
Build finished SUCCESSFULLY. See http://jenkins.elasticbox.com/job/chi_pull_request/580/ for more details.
